### ISSUE TITLE
 Star TSP100III futurePRNT(LAN) not printing

### ISSUE THREAD

**Issue by peterkuo02** [mike42/escpos-php#569]
@mike42
Thanks for the wonderful project.
I've tried to run the example/text-size.php, the printer did trigger the drawer but the data is not printing.
I also ran demo.php with the correct printer setting and it was not printing as well.
The printer is Star TSP100III futurePRNT(LAN)

**Comment by mike42** [comment]
Does your printer support ESC/POS, and have you checked that it's enabled? ([example for a different Star printer](http://www.starmicronics.com/absolutefm/absolutefm/afmviewfaq.aspx?faqid=72))

**Comment by peterkuo02** [comment]
I've followed that instruction and checked the printer setting of ESC/POS emulation is checked.
Since I was checking the setting on my windows desktop, but I was testing to print on the Macbook with the command : 

> php text-size.php 


I just realized that might be the issue and probably need to try it again to do all the work on the same machine. 
I will keep you updated. Thanks.

**Comment by peterkuo02** [comment]
Sorry it took a while to run another test.
I've setup the server and call the script from windows desktop, with the printer setting( Enable ESC/POS Routing) and the register drawer is open but the data is not printing.

**Comment by mike42** [comment]
Hm, so the basic problem in this case is still that either the printer is not getting ESC/POS, or that it doesn't understand what we're sending.

We have a lot of Star users who don't have this problem, so can you please tell us more about your setup?

- How the printer connects
- Operating systems involved
- How you have set up the printer in the OS (Should be `Generic / Text only`!)
- Where this setting for ESC/POS emulation is set (it needs to be a persistent setting)
- A command that you can run to deliver raw data. If it's a shared printer on Windows, this is usually:

```
echo "Hello World" > foo.bin
cp foo.bin \\computer\printer
```

And the next debugging steps to take:

- If you can't get this plaintext to your printer, then you have other problems. Debug with your vendor until this works.
- If this works, you need an ESC/POS sample file to test that you are in emulation mode. If you don't have any, you can use `demo.bin` from [here](https://github.com/mike42/escpos-php/tree/development/test/integration/resources/output) and send it via the same method. Debug with your vendor until this works.

Once you have a working command to get raw data to your printer, and your printer is handling ESC/POS, then you are ready to use this library.

There is a Star example of loading a `CapabilityProfile` in the README, and a `WindowsPrintConnector` example to match, though it looks like you've already got transport figured out from the PHP size.

**Comment by peterkuo02** [comment]
Here is the whole environment of the system...

1. The printer is connected to the router directly
2. apache is running on Mac, connected wirelessly to the router, run the script throw browser on a Windows desktop, wired to the router
The ECO/Pos mode is set as the manual showed. I was wondering if I did not set it correctly. I might need to figure out how to set it permanently  .
  

-

**Comment by mike42** [comment]
> The printer is connected to the router directly

Is it an Ethernet printer? Can you share a self-contained code snippet showing how you are using it from PHP so that we can see how it's being set up?

> The ECO/Pos mode is set as the manual showed.

The printer failing to print a simple job usually means it doesn't understand you, so the emulation mode is important. See the debugging steps in my first response.

I'll marking this as 'invalid' until you can demonstrate that you have a setup that is capable of accepting ESC/POS. This is not a printer support forum, please get your printer working on the command-line before you try to use `escpos-php`.

**Comment by GianniCarlo** [comment]
Adding more info to this, as I found myself in this exact situation, having a Raspberry Pi 3 in the same network as the TSP100III futurPRNT(LAN).

Using [this guide](http://www.starmicronics.com/support/MannualFolder/common_tsp_linux_en.pdf) as a basis, I managed to register the printer via the CUPS admin interface (on the Pi), and successfully send a test page to the printer.

Using the `CupsPrintConnector`, the job is created (and completed) in the printer, but unfortunately no actual printing is done.

I've made sure the printer is in `ESC/POS` emulation mode following your previous comment @mike42 
> Does your printer support ESC/POS, and have you checked that it's enabled? ([example for a different Star printer](http://www.starmicronics.com/absolutefm/absolutefm/afmviewfaq.aspx?faqid=72))

So I guess the next thing would be to check the difference between how the payload is sent in CUPS for the test page, vs how it's sent with this library to CUPS.

Any pointers would be appreciated @mike42 . Thanks again for this library!
